cmake/modules/AddCephTest.cmake

```diff
@@ -23,6 +23,21 @@ function(add_ceph_test test_name test_path)
     PROPERTY TIMEOUT ${CEPH_TEST_TIMEOUT})
 endfunction()
 
+function(add_test_label test_name label)
+    get_property(loc_label TEST ${test_name} PROPERTY LABELS)
+    if("${loc_label}" STREQUAL "" )
+        set_tests_properties(${test_name} PROPERTIES LABELS "${label}")
```

Comment:
Isn't here an `add_tests_properties` function? So that no pre-existing labels are removed? Or is this additive?

Additionally, we may also set the [COST](https://cmake.org/cmake/help/v3.8/prop_test/COST.html), although it seems to me that it has weird semantics (the higher the cost the sooner the test is run) and/or [DEPENDS](https://cmake.org/cmake/help/v3.8/prop_test/DEPENDS.html) to run later test depending on other ones.

Anyway, this looks really interesting. I was also exploring [tox-tags](https://pypi.org/project/tox-tags/), which allows for filtered runs of tests. We may pass these LABELS down to the tox tests and allow, for example, to run all RGW-related tests across different suites (including dashboard ones).

Reply:
@epuertat 
`set_tests_properties` is the only function I could find, and it not additive.
Hence the `add_test_label` function above.

I tried faking COST by adding the "long" for those tests running > 100sec on my build platform.

Now what I really would like is to dictate the running order on Labels and run like: "long+short" . And then start with the long tests, and only fill free slots with short tests when all long tests are done. This way the testing has the shortest run time, without solving a bin-packing problem. ;-)
Looks like your COST could do that. Will have to look into that.

Getting LABELS to propagate would be cool, especially for those tests that seem to run "forever" and you can thus manipulate functionality.

Reply:
@tchaikov
I think that is a matter of taste. Using a function for this, allow easier future changes in (more) functionality. Something I can imagine is adding a `COST` setting if one of the labels is `long`.
Which is part of the dicussion below.
But as usual, our milage may vary.